example/Benchmark.cpp

```diff
@@ -0,0 +1,380 @@
+
+        //These matrices will be filled with the EOS values (overwriting for each trail)
+        //Then at the end of the program, we export the matrix to a .csv to later use for accuracy tests
+        std::vector<std::vector<Real>> sie_eos_ref(nRho, std::vector<Real>(nT)); 
+        std::vector<std::vector<Real>> sie_rt(nRho, std::vector<Real>(nT));
+        std::vector<std::vector<Real>> sie_rs(nRho, std::vector<Real>(nT));
+
+        std::vector<std::vector<Real>> P_eos_ref_T(nRho, std::vector<Real>(nT));
+        std::vector<std::vector<Real>> P_rt_T(nRho, std::vector<Real>(nT));
+        std::vector<std::vector<Real>> P_rs_T(nRho, std::vector<Real>(nT));
+
+        std::vector<std::vector<Real>> P_eos_ref_sie(nRho, std::vector<Real>(nT));
+        std::vector<std::vector<Real>> P_rt_sie(nRho, std::vector<Real>(nT));
+        std::vector<std::vector<Real>> P_rs_sie(nRho, std::vector<Real>(nT));
+
+        std::vector<std::vector<Real>> T_back_eos_ref(nRho, std::vector<Real>(nT));
+        std::vector<std::vector<Real>> T_back_rt(nRho, std::vector<Real>(nT));
+        std::vector<std::vector<Real>> T_back_rs(nRho, std::vector<Real>(nT));
+
+        // === Benchmark Loop ===
+        for (int rep = 0; rep < ntimes; ++rep) {
+                // Clear matrices
+                for (auto& row : sie_eos_ref) std::fill(row.begin(), row.end(), 0.0);
+                for (auto& row : sie_rt) std::fill(row.begin(), row.end(), 0.0);
+                for (auto& row : sie_rs) std::fill(row.begin(), row.end(), 0.0);
+
+                for (auto& row : P_eos_ref_T) std::fill(row.begin(), row.end(), 0.0);
+                for (auto& row : P_rt_T) std::fill(row.begin(), row.end(), 0.0);
+                for (auto& row : P_rs_T) std::fill(row.begin(), row.end(), 0.0);
+
+                for (auto& row : P_eos_ref_sie) std::fill(row.begin(), row.end(), 0.0);
+                for (auto& row : P_rt_sie) std::fill(row.begin(), row.end(), 0.0);
+                for (auto& row : P_rs_sie) std::fill(row.begin(), row.end(), 0.0);
+
+                for (auto& row : T_back_eos_ref) std::fill(row.begin(), row.end(), 0.0);
+                for (auto& row : T_back_rt) std::fill(row.begin(), row.end(), 0.0);
+                for (auto& row : T_back_rs) std::fill(row.begin(), row.end(), 0.0);
+
+
+                // == EOSPAC (_eos) model ==
+                auto t0 = high_resolution_clock::now(); //e(rho, T) 
+                for (int i = 0; i < nRho; ++i) { //replace these with a single for loop that loops through (i,j)
+                    for (int j = 0; j < nT; ++j) { 
+                        sie_eos_ref[i][j] = eos_ref.InternalEnergyFromDensityTemperature(rhos[i], temps[j]); } }
+                auto t1 = high_resolution_clock::now();
+                std::chrono::duration<double, std::micro> elapsed = t1 - t0; //define "elapsed"
+                time_sie_eos_ref_list.push_back(elapsed.count());
```

Comment:
This pattern is repeated a lot and could be a function with a signature something like
```c++
template<typename EOS_t>
double time_InternalEnergyFromDensityTemperature(const EOS_t eos,
    const std::vector<double> rhos, const std::vector<double> temps,
    std::vector<std::vector<double>> sie_values);
```
with one version for each type of lookup. There may be a way to inject the type of lookup as well, but there are some complications there.